src/main/java/de/tracetronic/jenkins/plugins/ecutest/test/config/ImportProjectConfig.java

```diff
@@ -38,34 +57,43 @@ public class ImportProjectConfig extends ImportConfig {
      * @param importMissingPackages specifies whether to import missing packages
      * @param credentialsId         the credentials id
      * @param timeout               the import timeout
+     * @param tmProjectId           the project ID in test management system
      */
     @DataBoundConstructor
     public ImportProjectConfig(final String tmsPath, final String importPath,
-                               final boolean importMissingPackages, final String credentialsId, final String timeout) {
+                               final boolean importMissingPackages, final String credentialsId, final String timeout,
+                               final String tmProjectId) {
         super(tmsPath, importPath, credentialsId, timeout);
         this.importMissingPackages = importMissingPackages;
+        this.tmProjectId = StringUtils.trimToEmpty(tmProjectId);
     }
 
     public boolean isImportMissingPackages() {
         return importMissingPackages;
     }
 
+    public String getTmProjectId() {
+        return tmProjectId;
+    }
+
     @Override
     public ImportProjectConfig expand(final EnvVars envVars) {
         final String expTmsPath = envVars.expand(getTmsPath());
         final String expImportPath = envVars.expand(getImportPath());
         final String expCredentialsId = envVars.expand(getCredentialsId());
         final String expTimeout = EnvUtil.expandEnvVar(getTimeout(), envVars, String.valueOf(DEFAULT_TIMEOUT));
         return new ImportProjectConfig(expTmsPath, expImportPath, isImportMissingPackages(),
-            expCredentialsId, expTimeout);
+            expCredentialsId, expTimeout, getTmProjectId());
     }
 
     @Override
     public final boolean equals(final Object other) {
         boolean result = false;
         if (other instanceof ImportProjectConfig) {
             final ImportProjectConfig that = (ImportProjectConfig) other;
-            result = that.canEqual(this) && super.equals(that) && importMissingPackages == that.importMissingPackages;
+            result =
+                that.canEqual(this) && super.equals(that) && importMissingPackages == that.importMissingPackages
+                    && StringUtils.trimToEmpty(tmProjectId).equals(StringUtils.trimToEmpty(that.tmProjectId));
```

Comment:
usage of StringUtils here shouldn't be necessary because of usage in constructor